src/javascripts/components/copy.mjs

```diff
@@ -19,34 +20,50 @@ class Copy {
 
     this.$button = document.createElement('button')
     this.$button.className = 'app-copy-button'
-    this.$button.setAttribute('aria-live', 'assertive')
     this.$button.textContent = 'Copy code'
 
+    this.$status = document.createElement('span')
+    this.$status.className = 'govuk-visually-hidden'
+    this.$status.setAttribute('aria-live', 'assertive')
+
+    this.$module.insertAdjacentElement('beforebegin', this.$status)
     this.$module.insertAdjacentElement('beforebegin', this.$button)
-    this.copyAction()
+
+    const $clipboard = new ClipboardJS(this.$button, {
+      target: (trigger) => trigger.nextElementSibling
+    })
+
+    $clipboard.on('success', (event) => this.successAction(event))
+    $clipboard.on('error', (event) => this.resetAction(event))
   }
 
   /**
-   * Set up button copy action
+  * Copy to clipboard success
+  *
+  * @param {import('clipboard').Event} event
+  */
+  successAction(event) {
+    this.$button.textContent = this.$status.textContent = 'Code copied'
+      // Reset button after 5 seconds
+    this.resetAction(event, 5000)
```

Comment:
Thanks for the tweaks by the way. Hope it worked? I'll give it a test run

Another little indentation fix here:

```suggestion

    // Reset button after 5 seconds
    this.resetAction(event, 5000)
```